### PR TITLE
main: use friendly filename in stdin read error

### LINKED ISSUE #726: bug: stdin read error message uses wrong name
### name

Bug report

### about

`io.ReadAll` error message uses `infile.Name()` instead of the `filename` variable

### title

stdin read error message uses wrong name

### labels

kind/bug

### warning

No response

---

## Bug Description

In `internal/main.go`, when `io.ReadAll` fails on line 127, the error message is formatted using `infile.Name()`:

```go
// internal/main.go, line 127 (commit cb34e120)
fail("failed to read %s: %v\n", infile.Name(), err)
```

https://github.com/coreos/butane/blob/cb34e120e5267bfd5bdfa83fa8c2e44e06dedda2/internal/main.go#L127

When no `--input` flag is given, `infile` is assigned `os.Stdin` (line 113). `os.Stdin.Name()` returns `"/dev/stdin"` on Linux. However, the `filename` variable is set to `"<stdin>"` (line 114) for exactly this case, and updated to the real path only when `--input` is provided (line 122):

```go
infile := os.Stdin
filename := "<stdin>"
if input != "" {
    var err error
    infile, err = os.Open(input)
    if err != nil {
        fail("failed to open %s: %v\n", input, err)
    }
    defer infile.Close()
    filename = input
}
```

The error on line 119 and the `TranslateBytes` call on line 132 both use `filename` or `input` directly. Line 127 is the only site that diverges by calling `infile.Name()`.

## Current behavior

When `io.ReadAll` fails reading from stdin, the error reads:

```
failed to read /dev/stdin: <error>
```

## Expected behavior

The error should use the same friendly identifier used everywhere else in the function:

```
failed to read <stdin>: <error>
```

## Evidence

`os.Stdin.Name()` returns `"/dev/stdin"` on Linux (this is the value returned by the underlying file descriptor's name, not the user-facing label). The `filename` variable was explicitly initialized to `"<stdin>"` to present a human-readable name in messages, as is already done for the `TranslateBytes` report path on line 132.

## Release checklist

No response

## Tagging

No response

## Fedora packaging

No response

## GitHub release

No response

## Quay release

No response

### PR DESCRIPTION
**Pull request by deepak0x**
## Summary

In `internal/main.go`, the `io.ReadAll` failure path formatted its error with `infile.Name()`, which returns `/dev/stdin` on Linux instead of the friendly `<stdin>` label used everywhere else in the function (and in the `TranslateBytes` report path).

## Changes

Extracted the input-reading logic into a `readInput(input string) ([]byte, string, error)` helper. It returns the friendly filename (`<stdin>` when reading from stdin, the real path when `--input`/a positional file is given) so the read error message and the translation report stay consistent.

`internal/main.go` line 127 (now in `readInput`):

```go
fail("failed to read %s: %v\n", filename, err)
```

`filename` is now `<stdin>` for the stdin case instead of `/dev/stdin`.

## Verification

Added `internal/main_test.go` covering `readInput`:
- reading from stdin yields content and the `<stdin>` filename
- reading a real file yields content and the real path as filename
- a missing file returns an error with the requested path as filename

`./test` passes (gofmt, go vet, full test suite, doc validation).

Before: `failed to read /dev/stdin: <error>`
After:  `failed to read <stdin>: <error>`

Fixes: #726

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of input from standard input and specified files.
  * Added clearer error handling for missing or unreadable files.

* **Tests**
  * Added coverage for standard input, file input, missing files, returned filenames, and failed reads.

### CODE REVIEW

File: internal/main_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	if err == nil {
+		t.Fatalf("expected error for missing file, got nil")
+	}
+	if data != nil {
+		t.Errorf("expected nil data, got %q", data)
+	}
+	if filename != missing {
+		t.Errorf("expected filename %q, got %q", missing, filename)
+	}
+}
```
[the remark this answers is not in the document]
Comment (deepak0x) [reply internal/main_test.go]: Addressed: consolidated the readInput tests into a single table-driven TestReadInput with subtests (stdin, empty stdin, file, missing file) and added the empty-stdin edge case as suggested. Force-pushed the update.

**Comment by deepak0x** [comment]
Closing in favor of coreos/ignition#2293. Butane is merging into Ignition (see coreos/ignition#2281), so the fix for the stdin read error now lives there. ignition#2293 ports this exact change, including the readInput helper and the main_test.go table tests, so this Butane PR is no longer needed.
